sys/stdio_udp/stdio_udp.c

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (C) 2023 ML!PA Consulting GmbH
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     sys
+ * @{
+ *
+ * @file
+ * @brief       STDIO over UDP implementation
+ *
+ * This file implements STDIO via a UDP that can be used with e.g. netcat:
+ *
+ *      nc -u fe80::7837:fcff:fe7d:1aaf%tapbr0 2323
+ *
+ * @warning     This is entirely unsecured, only use this for debugging in
+ *              an isolated network!
+ *
+ * @author      Benjamin Valentin <benjamin.valentin@ml-pa.com>
+ *
+ * @}
+ */
```

Comment:
This is a bit of wasted effort, considering that `*.c` files are not visible to Doxygen. Maybe move this to a `doc.txt` instead?

IMO it we should also have a new Doxygen group for this. We mostly have one Doxygen group per module. IMO we should aim to having one Doxygen group for every module.

Reply:
I took the opportunity to also properly sort/document the other stdio implementations.

If you prefer this as a separate PR, I can split it out.